src/lib/journey/journey.store.ts

```diff
@@ -83,6 +85,15 @@ export function initialize(initOptions?: StepOptions): JourneyStore {
       ...nextOptions,
     };
 
+    // These options are reserved only for restarting a journey after failure
+    if (initOptions || nextOptions) {
+      restartOptions = {
+        // Prioritize next options over initialize options
+        ...initOptions,
+        ...nextOptions,
+      };
+    }
```

Comment:
This is just to enable proper journey restarting.

Reply:
Nice that we have a way to control this now - glad we've implemented this.